cpp/src/slice2cpp/Gen.cpp

```diff
@@ -242,33 +242,47 @@ namespace
         return stName;
     }
 
+    /// Returns the fully scoped name of the provided Slice definition, but scope separators will be replaced with
+    /// '_' characters (converting a scoped identifier into a single identifier).
+    string flattenedScopedName(const ContainedPtr& p)
+    {
+        string s = p->mappedScoped();
+        string::size_type pos = 0;
+        while ((pos = s.find("::", pos)) != string::npos)
+        {
+            s.replace(pos, 2, "_");
+        }
+        return s;
+    }
+
     string condMove(bool moveIt, const string& str) { return moveIt ? string("::std::move(") + str + ")" : str; }
 
+    /// Ensures that there is no collision between 'name' and any of the parameters in the provided 'params' list.
+    /// If a collision exists, we return 'name' with an underscore appended to it. Otherwise we return 'name' as-is.
     string escapeParam(const ParameterList& params, const string& name)
     {
-        string r = name;
         for (const auto& param : params)
         {
-            if (fixKwd(param->name()) == name)
+            if (param->mappedName() == name)
             {
-                r = name + "_";
-                break;
+                return name + "_";
             }
         }
-        return r;
+        return name;
     }
 
     /// Returns a doxygen formatted link to the provided Slice identifier.
+    /// TODO we need to add a way for the doc-comment generation to use 'cpp' identifier!
```

Comment:
It doesn't make sense for C++ doc-comment links to link to Slice identifiers.

Reply:
I agree, but this is going to need to be it's own PR.

To fix this, the doc-comment parsing code need to ability to look up Slice definitions by name, and check metadata on them. Right now it has no way to access any of this. And adding this functionality will affect all the languages since this comment-parsing code is shared.

This is a good thing to do anyways, so we can report broken links.
But it's going to a sub-project in-of-itself.